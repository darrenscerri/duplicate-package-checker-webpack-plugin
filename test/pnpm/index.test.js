var webpack = require("webpack");
var assert = require("assert");
var MakeConfig = require("./webpack.config");

describe("Pnpm dependency tree", function() {
  it("should not output warnings when no duplicates exist", function(done) {
    webpack(MakeConfig("entry.js"), function(err, stats) {
      assert(stats.compilation.warnings.length === 0);
      done();
    });
  });

  it("should output a warning when package 'a' is duplicated with the same version", function(done) {
    webpack(MakeConfig("entry.dupe.same-version.js"), function(err, stats) {
      assert(stats.compilation.warnings.length === 1);
      expect(stats.compilation.warnings[0].message).toMatchSnapshot();
      done();
    });
  });

  it("should output a warning when package 'a' is duplicated with a different version", function(done) {
    webpack(MakeConfig("entry.dupe.different-version.js"), function(
      err,
      stats
    ) {
      assert(stats.compilation.warnings.length === 1);
      expect(stats.compilation.warnings[0].message).toMatchSnapshot();
      done();
    });
  });
});
