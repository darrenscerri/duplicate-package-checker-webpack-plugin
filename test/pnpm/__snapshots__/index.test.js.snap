// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pnpm dependency tree should output a warning when package 'a' is duplicated with a different version 1`] = `
"a
[0m  Multiple versions of [0m[32m[1ma[22m[39m[37m found:[39m
[37m[39m    [32m[1m1.0.0[22m[39m [37m[1m./~/a[22m[39m
    [32m[1m1.1.0[22m[39m [37m[1m./~/c/~/a[22m[39m

[37m[1mCheck how you can resolve duplicate packages: [22m[39m
https://github.com/darrenscerri/duplicate-package-checker-webpack-plugin#resolving-duplicate-packages-in-your-bundle
"
`;

exports[`Pnpm dependency tree should output a warning when package 'a' is duplicated with the same version 1`] = `
"a
[0m  Multiple versions of [0m[32m[1ma[22m[39m[37m found:[39m
[37m[39m    [32m[1m1.0.0[22m[39m [37m[1m./~/b/~/a[22m[39m
    [32m[1m1.0.0[22m[39m [37m[1m./~/a[22m[39m

[37m[1mCheck how you can resolve duplicate packages: [22m[39m
https://github.com/darrenscerri/duplicate-package-checker-webpack-plugin#resolving-duplicate-packages-in-your-bundle
"
`;
